src/common/test-pipe.c

```diff
@@ -319,6 +330,107 @@ test_pid (TestCase *tc,
   g_assert_cmpuint (pid, ==, check);
 }
 
+static void
+on_pressure_set_throttle (CockpitPipe *pipe,
+                          gboolean throttle,
+                          gpointer user_data)
+{
+  gint *data = user_data;
+  g_assert (user_data != NULL);
+  *data = throttle ? 1 : 0;
+}
+
+static void
+test_pressure_queue (TestCase *tc,
+                     gconstpointer data)
+{
+  MockEchoPipe *echo_pipe = (MockEchoPipe *)tc->pipe;
+  gint throttle = -1;
+  GBytes *sent;
+  gint i;
+
+  g_signal_connect (tc->pipe, "pressure", G_CALLBACK (on_pressure_set_throttle), &throttle);
+  sent = g_bytes_new_take (g_strnfill (10 * 1000, '?'), 10 * 1000);
+
+  /* Sent this a thousand times */
+  for (i = 0; i < 1000; i++)
+    cockpit_pipe_write (tc->pipe, sent);
+
+  g_bytes_unref (sent);
+
+  /*
+   * This should have put way too much in the queue, and thus
+   * emitted the back-pressure signal. This signal would normally
+   * be used by others to slow down their queueing, but in this
+   * case we just check that it was fired.
+   */
+  g_assert_cmpint (throttle, ==, 1);
+  throttle = -1;
+
+  /*
+   * Now the queue is getting drained. At some point, it will be
+   * signaled that back pressure has been turned off
+   */
+  while (throttle == -1)
+    g_main_context_iteration (NULL, TRUE);
+
+  g_assert_cmpint (throttle, ==, 0);
+  g_assert_cmpint (echo_pipe->received->len, >, 10 * 1000);
```

Comment:
why 10*1000?

Reply:
I guess I wanted to say "are you sure that this can never be 0, and if you're sure, then shouldn't you also be sure that it would be more than one block?"